chapter05/spg/exercise2.pl

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+# There are no comments!
+my $data = {};
+my $sources = {};
+
+while ( <> ) {
+  my ( $source, $dest, $transfer ) = split;
+  $data->{$source}->{$dest} += $transfer;
+  $sources->{$source} += $transfer;
+}
+
+foreach my $source ( sort { $sources->{$b} <=> $sources->{$a} } keys $sources ) {
+
+  print sprintf( "Total for %-50s: %-10s\n", $source, $sources->{$source} );
```

Comment:
This is to make everything line up nicely, so the colon is set to 60 characters along the line.